storage/include/rocksdb/native_client.h

```diff
@@ -237,6 +237,17 @@ class NativeClient : public std::enable_shared_from_this<NativeClient> {
     return ::rocksdb::Slice{span.data(), span.size()};
   }
 
+  template <
+      typename Span,
+      std::enable_if_t<std::is_convertible_v<decltype(std::declval<Span>().size()), std::size_t> &&
+                           std::is_pointer_v<decltype(std::declval<Span>().data())> &&
+                           std::is_convertible_v<std::remove_pointer_t<decltype(std::declval<Span>().data())> (*)[],
+                                                 const uint8_t (*)[]>,
+                       int> = 0>
+  static ::rocksdb::Slice toSlice(const Span &span) noexcept {
```

Comment:
Nice!

Nit: Could we please update the comment at the top of the file that mentions NativeClient only accepts vector<char>, span<char> and so on to vector<char/uint8_t>, span<char/uint8_t>.